app/jobs/organization_event_job.rb

```diff
@@ -19,20 +19,28 @@ def perform(payload_body)
 
     return true unless @user
 
-    transfer_assignments if user_owns_assignments?
+    transfer_assignments if user_owns_any_assignments?
     @organization.users.delete(@user)
   end
   # rubocop:enable Metrics/AbcSize
 
+  def user_owns_any_assignments?
+    user_owns_assignments? || user_owns_group_assignments?
+  end
+
   def user_owns_assignments?
-    @organization.all_assignments.map(&:creator_id).include? @user.id
+    @organization.assignments.where(creator_id: @user.id).count.positive?
+  end
+
+  def user_owns_group_assignments?
+    @organization.group_assignments.where(creator_id: @user.id).positive?
   end
 
   def transfer_assignments
```

Comment:
The stuff in this method was pretty rough, nice refactor. This looks much better!